CountlyConnectionManager.m

```diff
@@ -414,15 +420,27 @@ - (void)endSession
 
 - (void)sendEvents
 {
-    [self sendEvents:false];
+    [self sendEvents:[Countly.user isLocallyCached]];
+}
+
+- (void)sendEvents:(BOOL) saveUser
+{
+    if(saveUser)
+    {
+        [Countly.user save];
+    }
+    else
+    {
+        [self sendEventsInternal:false];
+    }
 }
 
 - (void)attemptToSendStoredRequests
 {
-    [self sendEvents:true];
+    [self sendEventsInternal:true];
```

Comment:
this is the only place that is true as far as I see. We could extract user save logic to here directly.

Reply:
yes its the only place but we need to save queue in file after adding it in queue and before proceeding to queue.
If its in start or end of the code login than we can extract it here. but in current case it is not.

Reply:
It would be good for a function to do only one thing, like sendEvents would only send events not + saving user profiles. That is why I said it like it. If we extract the logic to another function or save where sendEvents used. 

If we put it inside sendEvents, it is easy for us to implement but it is hard for a newcomer or another one to read, or event it would be hard to debug. 

How I imagined:

function attempToSendBlabla
    sendEvents
    saveUserProperties
    sendRequests
    
    
It would be hard to manage maybe but it would be easy to see what is going on. So every place that is using this would become a Mediator